### DISCUSSION TITLE
Would it be possible to attach a canvas on a fabric js rectangle?

### DISCUSSION THREAD

**Discussion by dmdum** [fabricjs/fabric.js#7573] [Q&A] [answered]
Hello, I'm trying to draw a polygon inside a rectangle that the user also creates. Drawing the polygon works something similar to this: https://jsfiddle.net/ckitisak/3rwv1ae9/20/ 

However, after I've drawn the rectangle and attempt to draw the polygon on top of it, the drag feature of the rectangle just shows. I was thinking that maybe I can attach a canvas to the rectangle after it's created so that I can draw the polygon on top of it. I want the points of the polygon to refer to the rectangle instead of the bigger canvas the rectangle is in. Would that be possible? How would I go about doing that?

![Untitled](https://user-images.githubusercontent.com/95848438/148021620-eeb691f7-19ff-4987-84e3-2c4328f26871.png)

**Comment by dmdum** [comment, marked as answer]
@asturur so sorry to blatantly tag you like this but would you be able to share your thoughts on this?

**Reply by dmdum** [reply]
Was able to draw inside the rectangle by disabled the **evented** property of the rectangle.

**Reply by asturur** [reply]
Yes, if you need the rectangle to be a simple visual placeholder, disabling eventing is correct
